nupm/mod.nu

```diff
@@ -0,0 +1,16 @@
+use utils/dirs.nu [ DEFAULT_NUPM_HOME nupm-home-prompt ]
+
+export-env {
+    # Ensure that $env.NUPM_HOME is always set when running nupm. Any missing
+    # $env.NUPM_HOME during nupm execution is a bug.
+    if 'NUPM_HOME' not-in $env {
+        $env.'NUPM_HOME' = $DEFAULT_NUPM_HOME
+    }
```

Comment:
quick note: i think this could be simplified to
```nushell
$env.NUPM_HOME = $env.NUPM_HOME? | $DEFAULT_NUPM_HOME
```
:yum: 

not a requested change here though :relieved: